contracts/MultiPartyEscrow.sol

```diff
@@ -120,8 +156,9 @@ contract MultiPartyEscrow {
         
         //compose the message which was signed
         bytes32 message = prefixed(keccak256(abi.encodePacked(this, channelId, channel.nonce, amount)));
-        // check that the signature is from the channel.sender
-        require(recoverSigner(message, signature) == channel.sender);
+        // check that the signature is from the channel.sender or signer
+        address signAddress = ecrecover(message, v, r, s);
+        require(signAddress != address(0) && (signAddress == channel.sender || signAddress == channel.signer));
```

Comment:
I propose to require that message is signed only by the signer.
```
require(signAddress == channel.signer);
```
1. If sender wants to sign the messages by himself he should set signer == sender.
2. It makes logic simpler. 
3. I cannot image a situation in which it might be useful to sign the messages by both signer or sender.
4. Allowing sender OR signer to sign the message makes "defense" weaker which potentially opens new attack vectors.